tests/platform_tests/test_reboot.py

```diff
@@ -81,7 +82,26 @@ def reboot_and_check(localhost, dut, interfaces, xcvr_skip_list,
     logging.info("Append the latest reboot type to the queue")
     REBOOT_TYPE_HISTOYR_QUEUE.append(reboot_type)
 
-    check_interfaces_and_services(dut, interfaces, xcvr_skip_list, reboot_type=reboot_type)
+    try:
+        check_interfaces_and_services(dut, interfaces, xcvr_skip_list, reboot_type)
+    except Exception as e:
+        logging.error("Failing due to {}".format(e))
+        raise e
+    finally:
+        # If supervisor node is rebooted in chassis, linecards also will reboot.
+        # Check if all linecards are back up.
+        if dut.is_supervisor_node():
+            lc_reboot_type = "power off"
```

Comment:
1. can we also add check for wait_for_startup on the linecards?
2. can we check interfaces and services by reusing `check_interfaces_and_services`?
3. could you please explain why `lc_reboot_type = "power off"`
```
    if dut.is_supervisor_node():
        for lc in duthosts.frontend_nodes:
            wait_for_startup(lc, localhost, delay=10, timeout=300)
            check_interfaces_and_services(lc, interfaces, xcvr_skip_list,
                                          reboot_type=reboot_type)
```